File: Python/49. Group Anagrams .py
class Solution:
        def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
            res= defaultdict(list)
            for i in strs :
                count=[0] * 26 #hashmap 
                for c in i :
                    count[ord(c)-ord('a')]+=1 #key using ascii values 
                res[tuple(count)].append(i)
            return list(res.values())
                                                                                        

File: Java/49. Group Anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String,List<String>> res = new HashMap();
        for (String s:strs){
            char[] chars = s.toCharArray();
            Arrays.sort(chars);
            String temp = new String(chars);
            if (!res.containsKey(temp)){
                res.put(temp, new ArrayList<>());
            }
            res.get(temp).add(s);
        }

        return new ArrayList<>(res.values());
    }
}